### ISSUE TITLE
Bind legacy two-segment full-model execution to verified source provenance

### ISSUE THREAD

**Issue by azumag** [azumag/unzen#1182]
## Problem

The legacy same-machine verifier `LLM-proto/tools/verify_split_onnx.py::verify_split()` still opens the reference model directly with:

```python
ort.InferenceSession(str(full_model_path), providers=providers)
```

Its `unzen-real-two-segment-onnx` manifest already records `sourceModel.sha256` / external-data provenance, but `verify_split()` currently only takes a stable manifest snapshot and does not bind the full-model ORT load to the source generation recorded there. The newer multi-segment logits and KV-decode verifiers now close this boundary with the hard-link execution snapshot from #1179/#1181.

A direct reuse from `verify_split_onnx.py` is not currently possible without creating an import cycle: `verify_multi_segment_onnx.py` imports feed/comparison helpers from `verify_split_onnx.py`. The source execution snapshot should therefore be extracted to a dependency-neutral module (or equivalent non-circular shared boundary) before the legacy verifier adopts it.

## Acceptance criteria

- Move/refactor the source-provenance + execution-snapshot boundary needed by #1179/#1181 into a dependency-neutral helper without changing existing multi-segment report schemas or behavior.
- `verify_split()` validates `sourceModel` graph + declared external-data bytes/SHA/identity before reference execution and opens the full ORT session from the verified snapshot rather than the original pathname.
- Preserve stable symlinked source inputs and lexical external-data relative locations.
- Fail closed on same-byte/digest path retarget, different-inode replacement, and in-place graph/external-data mutation while the reference session can observe the source.
- Do not copy/embed multi-GiB external payloads or add a second full payload hash pass; retain the metadata-only hard-link/fingerprint strategy unless a better bounded-I/O approach is justified.
- Preserve provider selection, tolerances, boundary semantics, report schema, and sequential session loading for the legacy two-segment gate.
- Add focused regressions for snapshot-backed ORT path, source retarget/mutation rejection, and dependency/import-cycle safety.
- Update `docs/source-model-execution-snapshot.md` to cover the legacy verifier once implemented.

Host-side numerical-evidence integrity only. No production deployment, credentials, billing, #908 publication-concurrency decision, or #877 legacy deadline-policy change.

Related: #1178/#1179, #1180/#1181, #167.
